project_weather_app/database.py

```diff
@@ -1,26 +1,55 @@
+"""
+The WeatherDatabase object manages the low level interaction with the database.
+"""
 import sqlite3
 from location import Location
+from util import get_file_path
+import os.path
 
 class WeatherDatabase():
     """
     Create database for storing weather time temperature
     """
     def __init__(self, db_file: str) -> None:
-        self.conn = sqlite3.connect(db_file)
-        self.cursor = self.conn.cursor()
-        self.cursor.execute('''
-            CREATE TABLE IF NOT EXISTS weather (
-                id INTEGER PRIMARY KEY AUTOINCREMENT,
-                longitude REAL,
-                latitude REAL,
-                time TEXT,
-                temperature REAL
-            );
-        ''')
+        # get complete path to the file
+        filepath = get_file_path(db_file, calling_file=__file__)        
+        
+        # check if db_file exist
+        if not os.path.exists(filepath):
+            print("Database not found: ", filepath)
+            self.conn = None # for destructor in case file does not exist.
+            exit(1)
```

Comment:
this is OK, usually for those cases you want to utilize exceptions, potentially creating custom exceptions for your individual edgecases, since those can be handled better than just force-quitting your application